### PR TITLE
NetBSD support

### PR DESCRIPTION
**Pull request by danielskowronski**
I've added (very) experimental support for NetBSD - highly portable BSD distribution.

Only osdep/*EthernetTap needed to be changed (plus some adding \_\_NetBSD\_\_ in some places in code and custom makefile). As on older OS X getifmaddrs() and related are missing in NetBSD kernel so I've added them from FreeBSD. Implementation of creating tap device is quick and dirty but for now - it works.

### CODE REVIEW

**Comment by danielskowronski** [comment]
Tested with `NetBSD 7.0.2 (GENERIC.201610210724Z) i386` and `gcc version 4.8.4 (nb2 20150115)`

File: osdep/NetBSDEthernetTap.cpp [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+				int exitcode = -1;
+				::waitpid(cpid,&exitcode,0);
+			} else throw std::runtime_error("fork() failed");
+
+			cpid = (long)vfork();
+			if (cpid == 0) {
+				string tmp;
+				sprintf((char*)tmp.c_str(), "%d", i);
+				string minor = tmp.c_str();
+				::execl("/sbin/mknod","/sbin/mknod",devpath,"c","169",minor.c_str(),(const char *)0);
```
Comment (krytarowski) [inline osdep/NetBSDEthernetTap.cpp]: `cd /dev && ./MAKEDEV tap`

File: osdep/NetBSDEthernetTap.cpp [hunk 2]
```diff
@@ -0,0 +1,484 @@
...
+	} else if (cpid > 0) {
+		int exitcode = -1;
+		::waitpid(cpid,&exitcode,0);
+	}
+
+	cpid = (long)vfork();
+	if (cpid == 0) {
+		string tmp="/dev/";
+		tmp+=_dev.c_str();
+		::execl("/bin/rm","/bin/rm",tmp.c_str(),(const char *)0);
```
Comment (krytarowski) [inline osdep/NetBSDEthernetTap.cpp]: `unlink(2)`?

File: Makefile [hunk 3]
```diff
@@ -16,3 +16,7 @@ endif
 ifeq ($(OSTYPE),OpenBSD)
 	include make-freebsd.mk
 endif
+
+ifeq ($(OSTYPE),NetBSD)
+	include make-netbsd.mk
```
Comment (krytarowski) [inline Makefile]: Is it possible to unify `make-freebsd.mk` and `make-netbsd.mk` into `make-bsd.mk`?

**Comment by adamierymenko** [comment]
Hmm... will have to see if this works against dev. Thanks!

**Comment by adamierymenko** [comment]
Now that OpenBSD is in `dev` I think we will have to look at this. Will have to be ported to `dev`.

File: Makefile (hunk 3 above)
Comment (adamierymenko) [inline Makefile]: In `dev` we now have a make-bsd which should do this.

**Comment by adamierymenko** [comment]
We will have to make a NetBSD CI VM on our VM host too.

**Comment by krytarowski** [comment]
I'm looking forward to it.

**Pull request merged**